src/sql/workbench/common/actions.ts

```diff
@@ -257,6 +257,10 @@ export class BackupAction extends Task {
 			if (serverInfo && serverInfo.isCloud && profile.providerName === mssqlProviderName) {
 				return accessor.get<INotificationService>(INotificationService).info(nls.localize('backup.commandNotSupported', 'Backup command is not supported for Azure SQL databases.'));
 			}
+
+			if (!profile.databaseName && profile.providerName === mssqlProviderName) {
```

Comment:
@alanrenmsft i am adding the provider name to ensure it doesnt affect pgsql - please let me know if something else is needed here